src/main/java/org/opentripplanner/model/plan/Itinerary.java

```diff
@@ -347,4 +255,227 @@ public String toStr() {
+  }
+
+  /**
+   * How much elevation is gained, in total, over the course of the trip, in meters. See
+   * elevationLost.
+   */
+  public Double getElevationGained() {
+    return elevationGained;
+  }
+
+  public void setElevationGained(Double elevationGained) {
+    this.elevationGained = DoubleRounder.roundTo2Decimals(elevationGained);
+  }
+
+  /**
+   * If a generalized cost is used in the routing algorithm, this should be the total cost computed
+   * by the algorithm. This is relevant for anyone who want to debug an search and tuning the
+   * system. The unit should be equivalent to the cost of "one second of transit".
+   * <p>
+   * -1 indicate that the cost is not set/computed.
+   */
+  public int getGeneralizedCost() {
+    return generalizedCost;
+  }
+
+  public void setGeneralizedCost(int generalizedCost) {
+    this.generalizedCost = generalizedCost;
+  }
+
+  /**
+   * This is the transfer-wait-time-cost. The aim is to distribute wait-time and adding a high
+   * penalty on short transfers. Do not use this to compare or filter itineraries. The filtering on
+   * this parameter is done on paths, before mapping to itineraries and is provided here as
+   * reference information.
+   * <p>
+   * -1 indicate that the cost is not set/computed.
+   */
+  public int getWaitTimeOptimizedCost() {
+    return waitTimeOptimizedCost;
+  }
+
+  public void setWaitTimeOptimizedCost(int waitTimeOptimizedCost) {
+    this.waitTimeOptimizedCost = waitTimeOptimizedCost;
+  }
+
+  /**
+   * This is the transfer-priority-cost. If two paths ride the same trips with different transfers,
+   * this cost is used to pick the one with the best transfer constraints (guaranteed, stay-seated,
+   * not-allowed ...). Do not use this to compare or filter itineraries. The filtering on this
+   * parameter is done on paths, before mapping to itineraries and is provided here as reference
+   * information.
+   * <p>
+   * -1 indicate that the cost is not set/computed.
+   */
+  public int getTransferPriorityCost() {
+    return transferPriorityCost;
+  }
+
+  public void setTransferPriorityCost(int transferPriorityCost) {
+    this.transferPriorityCost = transferPriorityCost;
+  }
+
+  /**
+   * This itinerary has a greater slope than the user requested.
+   */
+  public boolean isTooSloped() {
+    return tooSloped;
+  }
+
+  public void setTooSloped(boolean tooSloped) {
+    this.tooSloped = tooSloped;
+  }
+
+  /**
+   * The maximum slope for any part of the itinerary.
+   */
+  public Double getMaxSlope() {
+    return maxSlope;
+  }
+
+  public void setMaxSlope(Double maxSlope) {
+    this.maxSlope = DoubleRounder.roundTo2Decimals(maxSlope);
+  }
+
+  /**
+   * If {@link org.opentripplanner.routing.api.request.RoutingRequest#allowKeepingRentedVehicleAtDestination}
+   * is set than it is possible to end a trip without dropping off the rented bicycle.
+   */
+  public boolean isArrivedAtDestinationWithRentedVehicle() {
```

Comment:
Maybe rename this to `hasArrivedAtDestinationWithRentedVehicle`

Reply:
Even if it is allowed according to the Java Bean standard, i do not like it. It has the drawback that you can not guess the name - is it "has" or "is". So, I tend to stick with the "main" form of Java Bean standard and use "isXYZ" even if it sounds awkward.

Reply:
That is a good point. I think this can remain as is.